src/platformsh.js

```diff
@@ -1,67 +1,447 @@
+     * to look up a route by ID.
+     *
+     * @param {string} id
+     *   The ID of the route to load.
+     * @return {object}
+     *   The route definition.  The generated URL of the route is added as a "url" key.
+     * @throws {Error}
+     *   If there is no route by that ID, an exception is thrown.
+     */
+    getRoute(id) {
+        for (let [url, route] of Object.entries(this.routes())) {
+            if (route.id == id) {
+                return route;
+            }
+        }
+
+        throw new Error(`No such route id found: ${id}`);
+    }
+
+    /**
+     * Retrieves the credentials for accessing a relationship.
+     *
+     * The relationship must be defined in the .platform.app.yaml file.
+     *
+     * @param {string} relationship
+     *   The relationship name as defined in .platform.app.yaml.
+     * @param {int} index
+     *   The index within the relationship to access.  This is always 0, but reserved
+     *   for future extension.
+     * @return {object}
+     *   The credentials array for the service pointed to by the relationship.
+     * @throws {Error}
+     *   Thrown if called in a context that has no relationships (eg, in build)
+     * @throws {RangeError}
+     *   If the relationship/index pair requested does not exist.
+     */
+    credentials(relationship, index = 0) {
+        if (!this.isValidPlatform()) {
+            throw new Error('You are not running on Platform.sh, so relationships are not available.');
+        }
+        if (this.inBuild()) {
+            throw new Error('Relationships are not available during the build phase.');
+        }
+        if (! this.relationshipsDef[relationship]) {
+            throw new RangeError(`No relationship defined: ${relationship}.  Check your .platform.app.yaml file.`);
+        }
+        if (! this.relationshipsDef[relationship][index]) {
+            throw new RangeError(`No index ${index} defined for relationship: ${relationship}.  Check your .platform.app.yaml file.`);
+        }
+
+        return this.relationshipsDef[relationship][index];
+    }
+
+    /**
+     * Returns a variable from the VARIABLES array.
+     *
+     * Note: variables prefixed with `env:` can be accessed as normal environment variables.
+     * This method will return such a variable by the name with the prefix still included.
+     * Generally it's better to access those variables directly.
+     *
+     * @param {string} name
+     *   The name of the variable to retrieve.
+     * @param {any} defaultValue
+     *   The default value to return if the variable is not defined. Defaults to null.
+     * @return mixed
+     *   The value of the variable, or the specified default.  This may be a string or an array.
+     */
+    variable(name, defaultValue = null) {
+        if (!this.isValidPlatform()) {
+            return defaultValue;
+        }
+
+        return this.variablesDef[name] || defaultValue;
```

Comment:
It's a bit of a slippery slope. `||` is not "if doesn't exist", but also "if false or undefined", so I'd rather have an explicit check there.